lib/ua.rb

```diff
@@ -0,0 +1,844 @@
+require "openstudio"
+
+# Set reference values for points, edges & surfaces (& subsurfaces) to
+# compute Quebec energy code (Section 3.3) UA' comparison (2021).
+#
+# @param [Hash] surfaces Preprocessed collection of TBD surfaces
+# @param [Hash] sets A model's PSI sets
+#
+# @return [Bool] Returns true if successful in generating UA' reference values
+def qc33(surfaces, sets)
```

Comment:
Basically 3 steps in UA calculations:
- set (reference) targets with respect to given code or standard (here, down to individual edge)
- analyse, then summarize (hash, minimally-formatted text ... useful for runner info & reports)
- generate reports (e.g. MD-formatted files)